code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -99,6 +107,9 @@
 	if (/obj/item/borg/upgrade/vtec in R.module.upgrades)
 		R.movement_speed_modifier -= vtec_bonus
 
+	for(var/obj/item/borg/upgrade/thing in R)
```

Comment:
I'm not keen on purely using the borg's contents for tracking which upgrades they have installed. Make a new list I'd say.